client/control/components/RecipeList.js

```diff
@@ -64,14 +64,62 @@ SwitchFilter.propTypes = {
 };
 
 class DisconnectedRecipeList extends React.Component {
-  propTypes = {
-    dispatch: pt.func.isRequired,
-    isFetching: pt.bool.isRequired,
-    recipeListNeedsFetch: pt.bool.isRequired,
-    recipe: pt.object.isRequired,
-    recipes: pt.array.isRequired,
+  /**
+   * Recipe metadata properties and associated labels to display
+   * @type {Object}
+   */
+  static ActionMetaData = {
+    'show-heartbeat': recipe => [
+      { label: 'Survey ID', value: recipe.arguments.surveyId },
+    ],
+  };
+
+  /**
+   * Given a recipe object, determines what type of recipe it is (based on its `action`),
+   * and then compiles an array of 'displayed metadata props' and their values. This array
+   * is saved on the recipe as `metaData`, and displayed in the 'Metadata'
+   *
+   * Beyond that, a string of metaData values is created, and attached to the
+   * recipe as the `searchData` property. This is used by the `Table` component
+   * to search/filter/sort the metaData.
+   *
+   * @param  {Object} Original recipe object
+   * @return {Object} Original recipe but with `metaData` and `searchData` properties added
+   */
+  static applyRecipeMetaData(recipe) {
+    const { action: recipeAction } = recipe;
+    const newRecipe = {
+      ...recipe,
+      // recipes should have empty metaData/searchData props,
+      // regardless if we set the values or not
+      metaData: [],
+      searchData: '',
+    };
+
+    // check if there are specific properties/labels we want to display
+    const requestedMetaProps = DisconnectedRecipeList.ActionMetaData[recipeAction];
+
+    // if we have a metadata definition to fill..
+    if (requestedMetaProps) {
+      const foundData = requestedMetaProps(newRecipe);
+
+      // add the data to the metaData collection
+      newRecipe.metaData = newRecipe.metaData.concat(foundData);
+    }
+
+    // update the searchdata string with whatever the values are
+    // (this is used for sorting/filtering/searching)
+    newRecipe.metaData.map(data => {
+      newRecipe.searchData += data.value;
+
+      // lint complains if we dont return
+      return data;
```

Comment:
The lint error could be avoided by using `.forEach` instead of `.map`. Alternatively, `for (const data of newRecipe.metaData) {` should work fine.

Separately does this need spaces between values? Maybe it doesn't matter, the search code probably doesn't care.